### PR TITLE
{bio}[GCCcore/10.20] ack v3.4.0 w/ Perl 5.32.0

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)
requires ~~#7276~~ (File::Next)

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2107.delcatty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/36213ffc5c8563aa639c094159a2cdf4 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3102.skitty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/3807b0c22d6eaf51d156b87f15c7550d for a full test report.

**Comment by bartoldeman** [comment]
Test report by @bartoldeman
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
lg-1r17-n06 - Linux centos 6.6, Intel(R) Xeon(R) CPU E5-2650 0 @ 2.00GHz, Python 2.6.6
See https://gist.github.com/5da0d04037ffdb128fdf339a47c3c1f8 for a full test report.

**Comment by bartoldeman** [comment]
Fails on `Can't locate File/Next.pm in @INC (you may need to install the File::Next module) (@INC contains: /localscratch/easybuild/build/ack/2.24/GCCcore-7.3.0-Perl-5.28.0/ack-2.24/blib/lib /localscratch/easybuild/build/ack/2.24/GCCcore-7.3.0-Perl-5.28.0/ack-2.24/blib/arch /localscratch/easybuild/software/Perl/5.28.0-GCCcore-7.3.0/lib/perl5/site_perl/5.28.0/x86_64-linux-thread-multi /localscratch/easybuild/software/Perl/5.28.0-GCCcore-7.3.0/lib/perl5/site_perl/5.28.0 /localscratch/easybuild/software/Perl/5.28.0-GCCcore-7.3.0/lib/perl5/5.28.0/x86_64-linux-thread-multi /localscratch/easybuild/software/Perl/5.28.0-GCCcore-7.3.0/lib/perl5/5.28.0) at t/resource-iterator.t line 7.
`

**Comment by bartoldeman** [comment]
ah, ok will do 7276 first :)

**Comment by bartoldeman** [comment]
Test report by @bartoldeman
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
lg-1r17-n06 - Linux centos 6.6, Intel(R) Xeon(R) CPU E5-2650 0 @ 2.00GHz, Python 2.6.6
See https://gist.github.com/a61fb6051ce0e38277f1fff88139bc26 for a full test report.

**Comment by bartoldeman** [comment]
Test report by @bartoldeman
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
lg-1r17-n06 - Linux centos 6.6, Intel(R) Xeon(R) CPU E5-2650 0 @ 2.00GHz, Python 2.6.6
See https://gist.github.com/485856b9d1bd34726a3b0b183ecfa867 for a full test report.

**Comment by bartoldeman** [comment]
@boegel it only works ok if i `unset BASH_ENV` before starting `eb`.
I'll try to do that in the easyconfig and see if it works there. Note:
`BASH_ENV=/software/CentOS-6/tools/lmod/lmod/init/bash` so lmod related

**Review by bartoldeman — DISMISSED** [review]
This should work with BASH_ENV set, which it does not for me. I am not sure how to unset it -- we have no `pretestopts` as far I can see, although disabling tests would do the trick of course.

**Comment by boegel** [comment]
@bartoldeman Do you know *why* having `BASH_ENV` set causes trouble?

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
b-cn0315.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/67b723cece1fc9567b4875fc1767a9d4 for a full test report.

**Comment by akesandgren** [comment]
The problem is probably not related to BASH_ENV, but rather that ack needs File::Next which is not part of that Perl version. It also has a PREREQ_PM that points to more things missing.

**Comment by boegel** [comment]
@akesandgren `File::Next` was added in ~~#7276~~.

What's the `PREREQ_PM` thing?

**Comment by boegel** [comment]
@akesandgren ping, need more info?

**Comment by boegel** [comment]
(re-triggering Travis checks)

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/ad06c21793db7bc3328762e243d27c57 for a full test report.

**Comment by smoors** [comment]
@boegelbot please test @ generoso

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node372.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/6fcca1aa8d30a61dbe8f961717daa23e for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3501.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/59cfb23e12f3d912f45dd745de02d6f9 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3409.kirlia.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 3.6.8
See https://gist.github.com/e22ad33cf6703e68fd32837b87d04420 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2707.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/c393deac1c39fb7a89f5dbb6dcbd2799 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3129.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/9338ebb449559755cfa7594df66838b2 for a full test report.

File: easybuild/easyconfigs/a/ack/ack-3.4.0-GCCcore-10.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+dependencies = [('Perl', '5.32.0')]
+
+options = {'modulename': 'App::Ack'}
+
+sanity_check_paths = {
+    'files': ['bin/ack', 'lib/perl5/site_perl/%(perlver)s/App/Ack.pm'],
+    'dirs': ['lib/perl5/site_perl/%(perlver)s/App/Ack'],
+}
+
+moduleclass = 'bio'
```
Comment (branfosj) [inline easybuild/easyconfigs/a/ack/ack-3.4.0-GCCcore-10.2.0.eb:25]:
```suggestion
moduleclass = 'tools'
```

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0211u38b.bear.cluster - Linux centos linux 8.2.2004, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/c320f2105a55b4bfbbb50d8949864820 for a full test report.

**Review by branfosj — APPROVED** [review]
lgtm

**Comment by zao** [comment]
Test report by @zao
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
eb-u1604 - Linux ubuntu 16.04, x86_64, Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz, Python 2.7.12
See https://gist.github.com/e3a9ae3fb54dc12f3798904d932b350b for a full test report.

**Comment by branfosj** [comment]
Going in, thanks @boegel!

**Pull request merged**